openedx/core/djangoapps/call_stack_manager/core.py

```diff
@@ -1,144 +1,203 @@
+                   str(frame[2]) + '\n\t' + str(frame[3]))
+
+    # Customize output of call stack
+    final_call_stack = ""
+    for frame in temp_call_stack:
+        final_call_stack += _print(frame)
+
+    def _should_get_logged(entity_name):
+        if not HALT_TRACKING:
+            if TRACK_FLAG and temp_call_stack not in STACK_BOOK[entity_name]:  # TRACK_FLAG False iff @donottrack
+                return True
+            else:
+                return False
+        else:
+            if inspect.isclass(entity_name):
+                if temp_call_stack not in STACK_BOOK[entity_name] and not issubclass(entity_name, tuple(HALT_TRACKING[-1])):
+                    return True
+                else:
+                    return False
+            else:  # Assumption : Everything other than "class" will be passed as string
+                if temp_call_stack not in STACK_BOOK[entity_name] and entity_name not in tuple(HALT_TRACKING[-1]):
+                    return True
+                else:
+                    return False
+
+    if _should_get_logged(entity_name):
+        STACK_BOOK[entity_name].append(temp_call_stack)
+        log.info("Logging new call stack number %s for %s:\n %s", len(STACK_BOOK[entity_name]),
+                 entity_name, final_call_stack)
 
 
 class CallStackMixin(object):
-    """ A mixin class for getting call stacks when Save() and Delete() methods are called
+    """ Mixin class for getting call stacks when save() and delete() methods are called
     """
 
     def save(self, *args, **kwargs):
         """
-        Logs before save and overrides respective model API save()
+        Logs before save() and overrides respective model API save()
         """
-        capture_call_stack(type(self))
+        if hasattr(self, 'model'):
+            capture_call_stack(self.model)
+        else:
+            capture_call_stack(type(self))
         return super(CallStackMixin, self).save(*args, **kwargs)
 
     def delete(self, *args, **kwargs):
         """
-        Logs before delete and overrides respective model API delete()
+        Logs before delete() and overrides respective model API delete()
         """
-        capture_call_stack(type(self))
+        if hasattr(self, 'model'):
+            capture_call_stack(self.model)
+        else:
+            capture_call_stack(type(self))
         return super(CallStackMixin, self).delete(*args, **kwargs)
 
 
 class CallStackManager(Manager):
-    """ A Manager class which overrides the default Manager class for getting call stacks
+    """ Manager class which overrides the default Manager class for getting call stacks
     """
     def get_query_set(self):
-        """overriding the default queryset API method
+        """ Override the default queryset API method
         """
-        capture_call_stack(self.model)
+        if hasattr(self, 'model'):
+            capture_call_stack(self.model)
+        else:
+            capture_call_stack(type(self))
```

Comment:
:white_check_mark: 

Removed it from `CallStackMixin`
I had to add this in `CallStackManager` because of `class Chunkingcallstackmanager`
In that case, `class Chunkingcallstackmanager` was logging its calls.